packages/did/src/DidResolver/DidResolver.ts

```diff
@@ -35,129 +31,133 @@ export async function resolve(
-      // If not, check whether the DID has been deleted or simply does not exist.
-      const isDeactivated = await queryDidDeletionStatus(did)
-      if (isDeactivated) {
-        return {
-          metadata: {
-            deactivated: true,
-          },
-        }
-      }
-      return null
+
+  const document = await Did.query(getFullDidUri(did))
+  if (type === 'full' && document) {
+    return {
+      document,
+      metadata: {
+        deactivated: false,
+      },
     }
-    case 'light': {
-      let document: DidDocument
-      try {
-        document = Did.parseDocumentFromLightDid(did, false)
-      } catch (cause) {
-        throw new SDKErrors.InvalidDidFormatError(did, {
-          cause: cause as Error,
-        })
-      }
-
-      const fullDid = await queryDetails(did)
-      // If a full DID with same subject is present, return the resolution metadata accordingly.
-      if (fullDid) {
-        return {
-          document,
-          metadata: {
-            canonicalId: fullDidUri,
-            deactivated: false,
-          },
-        }
-      }
-      // If no full DID document is found but the full DID has been deleted, return the info in the resolution metadata.
-      const isFullDidDeleted = await queryDidDeletionStatus(did)
-      if (isFullDidDeleted) {
-        return {
-          // No canonicalId and no document are returned as we consider this DID deactivated/deleted.
-          metadata: {
-            deactivated: true,
-          },
-        }
-      }
-      // If no full DID document nor deletion info is found, the light DID is un-migrated.
-      // Metadata will simply contain `deactivated: false`.
-      return {
-        document,
-        metadata: {
-          deactivated: false,
-        },
-      }
+  }
+
+  // If the full DID has been deleted (or the light DID was upgraded and deleted),
+  // return the info in the resolution metadata.
+  const isFullDidDeleted = await queryDidDeletionStatus(did)
+  if (isFullDidDeleted) {
+    return {
+      // No canonicalId and no details are returned as we consider this DID deactivated/deleted.
+      metadata: {
+        deactivated: true,
+      },
     }
-    default:
-      throw new SDKErrors.UnsupportedDidError(did)
+  }
+
+  if (type === 'full') {
+    return null
+  }
+
+  const lightDocument = Did.parseDocumentFromLightDid(did, false)
+  // If a full DID with same subject is present, return the resolution metadata accordingly.
+  if (document) {
+    return {
+      metadata: {
+        canonicalId: getFullDidUri(did),
+        deactivated: false,
+      },
+    }
+  }
+
+  // If no full DID details nor deletion info is found, the light DID is un-migrated.
+  // Metadata will simply contain `deactivated: false`.
+  return {
+    document: lightDocument,
+    metadata: {
+      deactivated: false,
+    },
+  }
+}
+
+type CompliantDidResolutionResult = Omit<DidResolutionResult, 'document'> & {
+  document?: DidDocument | { uri: DidUri }
```

Comment:
if that is meant to be fully DID compliant, then `uri` must be `id`

Reply:
It’s not meant to return a `CompliantDidDocument`

Reply:
phew, do you think we should have a function that's called `compliantResolve` but does not return a compliant did document?

Reply:
I’m open to naming suggestions :shrug: Returning a compliant document would require pulling in DidExporter, and I hesitate to do that.

Reply:
I came up with `strictResolve` and everyone likes it